mne/viz/eyetracking/heatmap.py

```diff
@@ -0,0 +1,185 @@
+    bin_width=10,
+    sigma=1,
+    cmap=None,
+    alpha=None,
+    vmin=None,
+    vmax=None,
+    axes=None,
+    show=True,
+):
+    """Plot a heatmap of eyetracking gaze data.
+
+    Parameters
+    ----------
+    epochs : mne.Epochs
+        The :class:`~mne.Epochs` object containing eyegaze channels.
+    width : int
+        The width dimension of the plot canvas. For example, if the eyegaze data units
+        are pixels, and the participant screen resolution was 1920x1080, then the width
+        should be 1920.
+    height : int
+        The height dimension of the plot canvas. For example, if the eyegaze data units
+        are pixels, and the participant screen resolution was 1920x1080, then the height
+        should be 1080.
+    bin_width : int
+        The number of eyegaze units per square bin that are used to create the heatmap.
+        Default is 10, meaning that if the eyegaze units are pixels, each bin is 10x10
+        pixels. See the appendix section of :ref:`tut-eyetrack-heatmap` for more
+        detail.
+    sigma : int | float
+        The amount of smoothing applied to the heatmap data. If ``None``,
+        no smoothing is applied. Default is 1.
+    cmap : matplotlib colormap | str | None
+        The :class:`~matplotlib.colors.Colormap` to use. Defaults to ``None``, meaning
+        the colormap will default to matplotlib's default.
+    alpha : int | float | None
+        The transparency value of the heatmap. If ``None``, the alpha value is set to 1,
+        meaning the heatmap colors are fully opaque. Default is ``None``.
+    vmin : float | None
+        The minimum value for the colormap. The unit is seconds, for dwell time
+        to the bin coordinate. If ``None``, the minimum value is set to the
+        minimum value of the heatmap. Default is ``None``.
+    vmax : float | None
+        The maximum value for the colormap. The unit is seconds, for dwell time
+        to the bin coordinate. If ``None``, the maximum value is set to the
+        maximum value of the heatmap. Default is ``None``.
+    axes : matplotlib.axes.Axes | None
+        The axes to plot on. If ``None``, a new figure and axes are created.
+        Default is ``None``.
+    show : bool
+        Whether to show the plot. Default is ``True``.
+
+    Returns
+    -------
+    fig : instance of matplotlib.figure.Figure
+        The resulting figure object for the heatmap plot.
+
+    Notes
+    -----
+    .. versionadded:: 1.6
+    """
+    from mne import BaseEpochs
+
+    _validate_type(epochs, BaseEpochs, "epochs")
+
+    # Find xpos and ypos channels. if future readers use different channel names than
+    # xpos/ypos, we will need a different way to identify these channels
+    xpos_indices = np.where(np.char.startswith(epochs.ch_names, "xpos"))[0]
+    ypos_indices = np.where(np.char.startswith(epochs.ch_names, "ypos"))[0]
+
+    data = epochs.get_data()
+    x_data = data[:, xpos_indices, :]
+    y_data = data[:, ypos_indices, :]
```

Comment:
Much more efficient to pass `picks=` to `get_data`. Also relying on channel names is unsafe because people can arbitrarily set. I would expect something like
```suggestion
    pos_picks = _picks_to_idx(epochs.info, "eyegaze")
    gaze_data = epochs.get_data(picks=pos_picks)
    x_data = gaze_data[np.where(...)[0]]
    ...
```
where the `np.where` is something having to do with `epochs.info['chs'][pick]['loc']` or `['coil_type']` or whatever tells you that it's a X gaze vs Y gaze. If it's *only* stored in the name currently this is something we should fix at the reader level.

Reply:
For Eyetracking objects, `epochs.info["chs"][pick]["loc"][4]` will be `-1` (x position) or `1` (y position).

And `epochs.info["chs"][pick]["loc"][4]` stores info on eye (-1 for left, 1 for right).

On that note, should we document this info somewhere (maybe in the `loc` parameter of mne.Info?). The only reason I know this information is because we talked about it during the `read_raw_eyelink` PR and I left some comments in the code.

Reply:
Let's put it here:

https://github.com/mne-tools/mne-python/blob/079c868240a898204bf82b2f1bf0e04cdee75da1/mne/_fiff/meas_info.py#L1187-L1192

And somewhere add a `# TODO: Add fNIRS convention to loc` nearby as we should also add it for those channels at some point. IIRC in that case it stores the nominal, source, detector position as well as the associated frequencies but I'd need to check.

Reply:
I took a whack at incorporating you suggestion of a smarter way to pick gaze channels in [6962df9](https://github.com/mne-tools/mne-python/pull/11798/commits/6962df96e9c01e556c050958d6281191103b7a9e), let me know if you think the code can be simplified there.